### PR TITLE
Use the Tumblr API to fetch posts.

### PR DESCRIPTION
**Pull request by jasonmc**
Did a refactor as I was looking to see about using this for Gnome desktops as well as OSX.

Missing options:
- small images (large only is default)
- image directory
- setting background from post id

### CODE REVIEW

**Comment by RichardLitt** [comment]
Dayum. @jasonmc, this is pretty clutch. @shadowhand, what do you think?

File: cabin-porn-it.py [hunk 1]
```diff
@@ -3,132 +3,89 @@
...
-    print('Image is large:', im)
-    setFile()
-  else: 
-    print('Image is too small:', im) # (width,height) tuple
-else: 
-  setFile()
+
+
+url = "https://api.tumblr.com/v2/blog/cabinporn.com/posts/photo"\
+      "?api_key=sZWRBqPLBfUJAsRY2Obu5SSioiwJasR0YGcgP5MSYulnQfpZyi"
```
Comment (shadowhand) [inline cabin-porn-it.py]: Where does this key come from?

**Comment by shadowhand** [comment]
:+1: I think it is pretty sexy!

**Comment by RichardLitt** [comment]
The README needs to be updated to match missing options, and I'm not sure about merging in a refactor if there aren't equivalent options. 

Does this work on Gnome, too?

**Comment by jasonmc** [comment]
Yeah, let me add the other options. Haven't added gnome support yet.

**Comment by RichardLitt** [comment]
Cool. Keep us updated!

**Comment by RichardLitt** [comment]
@jasonmc Ping?

**Pull request closed without merging**